core/rawdb/accessors_chain.go

```diff
@@ -192,8 +197,13 @@ func ReadHeaderRLP(db ethdb.Reader, hash common.Hash, number uint64) rlp.RawValu
 	// but when we reach into leveldb, the data was already moved. That would
 	// result in a not found error.
 	data, _ = db.Ancient(freezerHeaderTable, number)
-	if len(data) > 0 && crypto.Keccak256Hash(data) == hash {
-		return data
+	if len(data) > 0 {
+		header := new(types.Header)
```

Comment:
maybe extract this into a function... can be one inside the current function, but to extract common logic

Reply:
something like
```go
if len(data) > 0 && sameHeader(data, hash) {
  return data
}
```